tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,316 @@
+ * manually change these values in a MySQL shell when they are not hosting a
+ * LORIS on localhost.
+ *
+ * Finally, the script runs tools/resetpassword.php so that the default admin
+ * password with the Raisinbread data set is not used.
+ *
+ * In order to prevent accidental data loss, the script prompts the user to
+ * manually type the exact name of the database which will be affected.
+ *
+ * In normal cases, the database connection will be established via the
+ * credentials in the config file. However, if the database tables have already
+ * been deleted then this will not work. In this case it is still possible to
+ * import Raisinbread data if the user has properly set up a MySQL configuration
+ * file and provides the name of their LORIS database.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+require_once 'generic_includes.php';
+$info = <<<INFO
+This script is used by LORIS developers to DELETE DATA and replace
+them with new test data.
+
+
+INFO;
+
+echo $info;
+
+echo "\e[0;31m*** Executing this script will result in the LOSS OF DATA! ***\e[0m\n";
+
+$cwd = getcwd();
+if (substr_compare($cwd, 'tools', mb_strlen($cwd) - mb_strlen('tools')) !== 0) {
+    die('Please run this script from the tools/ directory.' . PHP_EOL);
+}
+
+try {
+    $dbInfo = $config->getSettingFromXML('database');
+
+    if (! $config->getSetting('dev')['sandbox']) {
+        throw new \LorisException(
+            "Config file indicates that this is not a sandbox. Aborting to " .
+            "prevent accidental data loss." . PHP_EOL
+        );
+    }
+    $dbname = $dbInfo['database'];
+    $host   = $dbInfo['host'];
+    // The admin user and password should be configured to a user with DROP
+    // and CREATE permissions for the database.
+    $username = $dbInfo['adminUser'];
+    $password = $dbInfo['adminPassword'];
+
+    $urlConfigSetting  = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+
+    echo <<<CONFIRMATION
+    Please type the database name `$dbname` to confirm you wish to drop tables
+    and import test data: 
+CONFIRMATION;
+
+    $input = trim(fgets(STDIN));
+    if ($input !== $dbname) {
+        die('Input did not match database name. Exiting.');
+    }
+
+} catch (\DatabaseException $e) {
+    echo 'Could not connect to the database in the Config file.' .
+        'It\'s possible that the LORIS tables have already been dropped.' .
+        PHP_EOL;
+    echo 'Continue attempting to install Raisinbread database? (y/N)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+    echo PHP_EOL . 'Please enter the name of your database:' . PHP_EOL;
```

Comment:
If you abort the script halfway through or some error occurs, it's possible that a DB exception can occur if LORIS tables have already been dropped. 

This prompt is a way to recover execution of the script when this happens.

Reply:
It's a fallback in case the credentials in the config file don't work.

This branch sets up the ability to connect using the command `mysql -A $dbname` below. This can happen if you have a MySQL configuration file set up. E.g. I have a user and password for MySQL configured in my home directory. If the DB info from the `project/config.xml` file doesn't work, this script prompts for the name of your LORIS test DB and tries to connect using your MySQL config instead.